docs/src/userguide/language_basics.rst

```diff
@@ -197,46 +152,81 @@ are supported:
 
                 ctypedef int* IntPtr
 
+.. _structs:
 
-You can create a C function by declaring it with :keyword:`cdef` or by decorating a Python function with ``@cfunc``:
+Structs, Unions, Enums
+----------------------
+
+In addition to the basic types, C :keyword:`struct`, :keyword:`union` and :keyword:`enum`
+are supported:
 
 .. tabs::
 
     .. group-tab:: Pure Python
 
-        .. code-block:: python
-
-            @cython.cfunc
-            def eggs(l: cython.ulong, f: cython.float) -> cython.int:
-                ...
+        .. literalinclude:: ../../examples/userguide/language_basics/struct.py
 
     .. group-tab:: Cython
 
-        .. code-block:: cython
+        .. literalinclude:: ../../examples/userguide/language_basics/struct.pyx
 
-            cdef int eggs(unsigned long l, float f):
-                ...
+Structs can be declared as ``cdef packed struct``, which has
+the same effect as the C directive ``#pragma pack(1)``:
 
-You can read more about them in :ref:`python_functions_vs_c_functions`.
+.. literalinclude:: ../../examples/userguide/language_basics/packed_struct.pyx
```

Comment:
The example is good but I think it's in the wrong place.

I think this might be too complicated for "language basics". I'm wondering if it should be in the "typed memoryviews" part of the documentation with a much shorter bit of text here:

> Structs can be declared as ``cdef packed struct``, which has
the same effect as the C directive ``#pragma pack(1)`` and removes empty space between members. They're most often used for working `[with memoryviews of structures](LINK GOES HERE)`. Pure python mode does not support packed structs.

Sorry for being indecisive. I'm just thinking that this is a bit overwhelming for new users.

Reply:
I put more complex example since we mention that it is often used with numpy arrays:

> The main use is that numpy structured arrays store their data in packed form

So I thought to put directly example how it can be used with numpy... But I can change it to simple example.

Reply:
OK. I had a another look. I removed the example left very basic example of definition. I put created there a link to memoryviews. I have left the text as it was before - I like having there link to wikipedia for more explanation about packed structs, etc.